src/Juvix/Data/Effect/ExactPrint/Base.hs

```diff
@@ -90,6 +91,31 @@ runExactPrint cs = reinterpretH (runPrivateStateAsDoc (initialBuilder cs)) handl
               _builderEnsureEmptyLine = st' ^. builderEnsureEmptyLine
             }
         return fx
+      RegionAlt regionModif alt (m1 :: Sem localEs x) (m2 :: Sem localEs x) -> do
+        st0 :: Builder <- set builderDoc mempty <$> get
+        let runner :: Sem (State Builder ': localEs) x -> Sem localEs (Builder, x)
+            runner = runState st0
+
+            helper :: Sem localEs x -> (forall w. Sem localEs w -> Sem r' w) -> Sem r' (Builder, x)
+            helper m unlift = unlift (impose runner handler m)
+
+            inner1 :: Sem r' (Builder, x)
+            inner1 = localSeqUnliftCommon locEnv (helper m1)
+
+            inner2 :: Sem r' (Builder, x)
+            inner2 = localSeqUnliftCommon locEnv (helper m2)
+        (st1 :: Builder, fx1) <- raise inner1
+        (st2 :: Builder, fx2) <- raise inner2
+        doc' <- gets (^. builderDoc)
+        put
+          Builder
```

Comment:
It should be clarified in the RegionAlt definition that m1 and m2 are expected to span the same region and thus consume the same comments. Otherwise they might get lost or repeated.
Maybe we could also add a sanity check? i.e. check that length of st1 and st2 comments match